js/recipe/metadata.js

```diff
@@ -0,0 +1,54 @@
+export function metadata(recipe) {
+
+  const data = {
+    publishedLabelSelector: "p.published-tag",
+    parentTagElementSelector: "span.tags",
+    parentTagElementLabel: "Tags: ",
+    tagNames: recipe.tags,
+    metaElements: [
+      {
+        selector: ".post-title",
+        data: recipe.name
+      },
+      {
+        selector: "time[itemprop='datePublished']",
+        data: new Date(recipe.datePublished).toDateString()
+      },
+      {
+        selector: "span.category",
+        data: `Serves: ${recipe.recipeYield} /`
+      }
+    ]
+  }
+
+  let publishedLabel = (published, selector) => {
+    if (published == false) {
+      document.querySelector(selector).style.display = "none";
+    }
+  }
+
+  let parentTagElement = (selector, label) => {
+    const element = document.querySelector(selector);
+    element.innerHTML = label;
+    return element
+  }
+
+  let populateMetaElements = (metaElements) => {
+    metaElements.forEach(({ selector, data }) => {
+      document.querySelector(selector).innerHTML = data;
+    });
+  }
+
+  let populateTags = (tagNames, parentElement) => {
```

Comment:
If we controlled the parent element it could look something like:

```js
const span = ({ className, children, innerHTML }) => ({
  element: "span",
  className,
  children,
  innerHTML,
})

const tags = tagNames.map(tagName => span({ innerHTML: `${tagName},&nbsp` }));
const parent = span({ className: "tags", children: tags });

const render = (target, { element, className, children, innerHTML }) => {
  const el = document.createElement(element);
  el.innerHTML = innerHTML; // TODO: only if innerHTML is not undefined
  el.className = className;
  el.innerHTML = children.forEach(child => { render(el, child); }); // if children defined - possibly only if innerHTML is not defined
  target.appendChild(el);
  return el;
}

render(document.body, parent);
```